components/stores/petStore.js

```diff
@@ -0,0 +1,82 @@
+import { create } from "zustand";
+import { persist, createJSONStorage } from "zustand/middleware";
+import initialMyPets from "@/lib/initialPet.js";
+import defaultMyPet from "@/lib/myPetTemplate.js";
+
+export const usePetStore = create(
+  persist(
+    (set, get) => ({
+      myPets: initialMyPets,
+      currentPet: null,
+      onSetCurrentPet: (myPet) => set({ currentPet: myPet }),
+      onAddPet: (newPet) => {
+        set((state) => ({
+          myPets: [...state.myPets, newPet],
+        }));
+      },
+      onUpdatePet: (updatedPet) => {
+        set({
+          myPets: get().myPets.map((myPet) =>
+            myPet.id === updatedPet.id ? updatedPet : myPet
+          ),
+        });
+      },
```

Comment:
```suggestion
        set((state) => ({
          myPets: state.myPets.map((myPet) =>
            myPet.id === updatedPet.id ? updatedPet : myPet
          ),
        }));
      },
```

We talked about it, avoid using get, use the state from inside every set function instead.

Reply:
mb, forgot this in the petStore